server/test/support/grpc_integration_case.ex

```diff
@@ -0,0 +1,59 @@
+defmodule BuffServerGrpc.IntegrationCase do
+  @moduledoc false
+  use ExUnit.CaseTemplate
+
+  using do
+    quote do
+      import BuffServerGrpc.IntegrationCase,
+        only: [
+          run_server: 2,
+          run_server: 3,
+          run_endpoint: 2,
+          run_endpoint: 3,
+          get_client: 2,
+          get_client: 3
+        ]
+    end
+  end
+
+  def run_server(servers, func, port \\ 0) do
+    {:ok, _pid, port} = GRPC.Server.start(servers, port)
+
+    try do
+      func.(port)
+    after
+      :ok = GRPC.Server.stop(servers)
+    end
+  end
+
+  def get_client(server, func, port \\ 0) do
```

Comment:
these should be macros or something.. i dont like having this extra level of indentation everywhere.